diff --git a/drivers/gpu/drm/drm_edid.c b/drivers/gpu/drm/drm_edid.c
index 6bb6337be920..70736fc3a6ba 100644
--- a/drivers/gpu/drm/drm_edid.c
+++ b/drivers/gpu/drm/drm_edid.c
@@ -579,7 +579,12 @@ static const struct drm_display_mode edid_est_modes[] = {
 		   DRM_MODE_FLAG_PHSYNC | DRM_MODE_FLAG_PVSYNC) }, /* 800x600@72Hz */
 	{ DRM_MODE("1152x864", DRM_MODE_TYPE_DRIVER, 108000, 1152, 1216,
 		   1344, 1600, 0,  864, 865, 868, 900, 0,
-		   DRM_MODE_FLAG_PHSYNC | DRM_MODE_FLAG_PVSYNC) }, /* 1152x864@75Hz */
+		   DRM_MODE_FLAG_PHSYNC | DRM_MODE_FLAG_PVSYNC) }, /* 1152x864@75Hz */
+	/* 0x59 - 480x800 Rienk special */
+	{ DRM_MODE("480x800", DRM_MODE_TYPE_DRIVER, 25000, 480, 490,
+			500, 515, 0, 800, 803, 806, 809, 0,
+			RM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC) },
+
 };
 
 struct minimode {
diff --git a/drivers/gpu/drm/ivip/intel_vip_of.c b/drivers/gpu/drm/ivip/intel_vip_of.c
index d7ce333de85e..79849a50f28d 100644
--- a/drivers/gpu/drm/ivip/intel_vip_of.c
+++ b/drivers/gpu/drm/ivip/intel_vip_of.c
@@ -113,7 +113,7 @@ static int intelvipfb_drm_setup(struct device *dev,
 		return -ENODEV;
 	}
 
-	drm->mode_config.min_width = 640;
+	drm->mode_config.min_width = 480;
 	drm->mode_config.min_height = 480;
 	drm->mode_config.max_width = max_w;
 	drm->mode_config.max_height = max_h;
